stonesoup/plotter.py

```diff
@@ -271,6 +274,51 @@ def plot_tracks(self, tracks, mapping, uncertainty=False, particle=False, track_
         else:
             self.ax.legend(handles=self.handles_list, labels=self.labels_list)
 
+    def plot_density(self, state_sequences: Iterable[StateMutableSequence],
+                     index: Union[int, None] = -1,
+                     mapping=(0, 2), n_bins=300, **kwargs):
+        """
+
+        Parameters
+        ----------
+        state_sequences : an iterable of :class:`~.StateMutableSequence`
+            Set of tracks which will be plotted. If not a set, and instead a single
+            :class:`~.Track` type, the argument is modified to be a set to allow for iteration.
+        index: int
+            Which index of the StateMutableSequences should be plotted.
+            Default value is '-1' which is the last state in the sequences.
+            index can be set to None if all indices of the sequence should be included in the plot
+        mapping: list
+            List of 2 items specifying the mapping of the x and y components of the state space.
+        n_bins : int
+            Size of the bins used to group the data
+        \\*\\*kwargs: dict
+            Additional arguments to be passed to pcolormesh function.
+        """
+
+        if index is None:  # Plot all states in the sequence
+            x = np.array([a_state.state_vector[mapping[0]]
+                          for a_state_sequence in state_sequences
+                          for a_state in a_state_sequence])
+            y = np.array([a_state.state_vector[mapping[1]]
+                          for a_state_sequence in state_sequences
+                          for a_state in a_state_sequence])
+        else:  # Only plot one state out of the sequences
+            x = np.array([a_state_sequence.states[index].state_vector[mapping[0]]
+                          for a_state_sequence in state_sequences])
+            y = np.array([a_state_sequence.states[index].state_vector[mapping[1]]
+                          for a_state_sequence in state_sequences])
+
+        # Evaluate a gaussian kde on a regular grid of n_bins x n_bins over data extents
+        k = kde.gaussian_kde([x, y])
```

Comment:
I think this raises an error when all values of x and y are the same. May want to catch that, as well as skip plotting if `state_sequences` is empty.

Reply:
Added raise errors for empty `state_sequences`  and when all x, y values are the same.